strongbox-web-core/src/main/java/org/carlspring/strongbox/domain/DirectoryContent.java

```diff
@@ -0,0 +1,89 @@
+package org.carlspring.strongbox.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DirectoryContent
+{   
+    Logger logger = LoggerFactory.getLogger(DirectoryContent.class);
+    
+    private List<FileContent> directories;
+    
+    private List<FileContent> files;
+    
+    @JsonCreator
+    public DirectoryContent( @JsonProperty("directories")  List<FileContent> directories,
+                             @JsonProperty("files") List<FileContent> files) 
+    {
+        this.directories = directories;
+        this.files = files;
+    }
+    
+    public DirectoryContent(Path dirPath) throws IOException
+    {
+        this.directories = new ArrayList<FileContent>();
+        this.files = new ArrayList<FileContent>();
+               
+        setDirectories(dirPath);
+        setFiles(dirPath);
+    }
+       
+    public void setDirectories(Path dirPath) throws IOException
+    {   
+        Files.list(dirPath)
```

Comment:
I think this logic should be implemented in some service instead of directly here.

Reply:
+1